#172062269 Salir del juego y eliminar basura

eljorgemo opened the pull request:
Está implementado que se pueda salir con el botón X de la ventana y con el botón Salir del menú de pausa (este último es provisional iirc).
Al parecer no queda basura ya. He conseguido arreglar lo de que petara al destruir el physicsWorld y el collisionsManager (al parecer los colliders se destruían a sí mismos y luego el mundo físico los volvía a intentar destruir, y otras cosas del estilo).
Probad que no pete, porfa. Intentad salir después de matar a alguien y cosas así (para que se queden colliders sueltos en el mundo), aunque yo lo he hecho y he arreglado los errores quitando un par de destrucciones.
P.D.: he añadido checkML a los .h que no lo tenían

JuanmaSalva committed:
La X de salir ya es funcional

Deja menos basura pero aun deja y no se q es xd pero me voy a cenar

eljorgemo committed:
basurilla

JuanmaSalva committed:
Merge branch 'master' into #172062269-salir

JuanmaSalva committed:
Deja menos basura

Pero aun deja. Lo que queda está anotado en la destructura del playstate.cpp q no se como resorverlo

eljorgemo committed:
incluidos checkML en muchos sitios

eljorgemo committed:
ya no peta y no deja basura (milagro?)

eljorgemo committed:
Se me había olvidado hacer que se pudiera salir desde el menú de pausa xd

EstebenR reviewed (changes requested):
El changeState no deberia borrar los demás estados, porque asi no hay que volver a crearlos cuando se vaya al menú de pausa o principal, el unico que se borra cuando el juego se está ejecutando es el playstate (cuando por ejemplo sales al menú y empiezas otra partida) los demás se borran al salir del juego

EstebenR commented:
Also el SDL_Game no debería estar revisando constantemente si se le ha pedido salir, hay que hacer un método publico para salir que las otras cosas llamen cuando se vaya a salir, no hacer poll constante

eljorgemo committed:
Ya no se hace delete de los estados al cambiar de estado

Lo he probado y no peta ni nada parecido, ni tampoco deja basura. @EstebenR era esto a lo que te referías?

eljorgemo committed:
Revert "Merge branch 'master' into #172062269-salir"

This reverts commit 40b0c7604bdce219829b6189c1abe24bb6f1286f, reversing
changes made to d8e4756584145d34c93891db1e292bd1d83caef6.

eljorgemo committed:
Ya no se hace poll continuamente para salir del juego

eljorgemo commented:
Me ha dado unos conflictos con master pero ya no los da y no he tocado nada (?) supongo que luego mostrará los conflictos again

EstebenR reviewed (changes requested):
buena liada, on hold hasta que se arregle

eljorgemo committed:
Ya no se hace poll continuamente para salir del juego

eljorgemo committed:
Merge de master
